### PR TITLE
Adding 2.190.x line

### PR DESCRIPTION
**Pull request by jglick**

### CODE REVIEW

**Comment by jglick** [comment]
Hmm, there are test failures but they are not getting reported:

```
java.lang.NullPointerException
	at hudson.tasks.junit.TestResultAction.getDataFile(TestResultAction.java:132)
	at hudson.tasks.junit.TestResultAction.load(TestResultAction.java:208)
	at hudson.tasks.junit.TestResultAction.getResult(TestResultAction.java:138)
	at hudson.tasks.junit.TestResultAction.getResult(TestResultAction.java:63)
	at hudson.tasks.test.AbstractTestResultAction.findCorrespondingResult(AbstractTestResultAction.java:252)
	at hudson.tasks.test.TestResult.getPreviousResult(TestResult.java:146)
	at hudson.tasks.junit.SuiteResult.getPreviousResult(SuiteResult.java:416)
	at hudson.tasks.junit.CaseResult.getPreviousResult(CaseResult.java:483)
	at hudson.tasks.junit.CaseResult.recomputeFailedSinceIfNeeded(CaseResult.java:411)
	at hudson.tasks.junit.CaseResult.freeze(CaseResult.java:644)
	at hudson.tasks.junit.SuiteResult.freeze(SuiteResult.java:462)
	at hudson.tasks.junit.TestResult.freeze(TestResult.java:748)
	at hudson.tasks.junit.JUnitResultArchiver.parseAndAttach(JUnitResultArchiver.java:178)
	at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:52)
	at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:25)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at …
```

which looks like `TestResultAction.run == null` for some reason. Not really sure what to make of that except something being seriously broken on the server. CC @olblak

**Comment by jglick** [comment]
I also saw

```
java.lang.NullPointerException
	at org.jenkinsci.plugins.workflow.cps.EnvActionImpl.getEnvironment(EnvActionImpl.java:88)
```

recently in this PR (build 2), again a case of an owner field not being updated from a `RunAction2`, suggesting a lower-level problem in Jenkins core or `workflow-job` or something. Do you see any warnings in the system log from `Run.onLoad` or anything like that?

**Comment by jglick** [comment]
Hoping the resolution of [INFRA-2283](https://issues.jenkins-ci.org/browse/INFRA-2283) unblocks this.

**Comment by jglick** [comment]
Finally got in results, with a bunch of [PCT failures](https://ci.jenkins.io/job/Tools/job/bom/job/PR-110/4/testReport/).

**Comment by basil** [comment]
> Finally got in results, with a bunch of [PCT failures](https://ci.jenkins.io/job/Tools/job/bom/job/PR-110/4/testReport/).

May I suggest incorporating this type of plugin compatibility testing into the official Jenkins LTS release process? In other words, why not do this type of testing _before_ an LTS line is released rather than after the fact? As an end user, I'd feel more comfortable upgrading to a given LTS release knowing that essential plugins (including Pipeline) have been tested with it.

**Comment by jglick** [comment]
> May I suggest

Yes!

I could have filed a draft PR like this one as soon as the LTS line is announced, just building against the weekly baseline. Unfortunately the way the `*/pom.xml` are set up means that if it were not merged promptly, every Dependabot PR would conflict with it, since Git only activates rename detection if the origin file no longer exists, and the way the BOM is set up, most of the content is always in the newest one. … I suppose I could just rename the directory for the latest line to be simply `bom-latest`, hmm. Will play with all this.

**Comment by basil** [comment]
> Finally got in results, with a bunch of [PCT failures](https://ci.jenkins.io/job/Tools/job/bom/job/PR-110/4/testReport/).

I wrote [PipelineTest.globalDecoratorAnnotator](https://ci.jenkins.io/job/Tools/job/bom/job/PR-110/4/testReport/hudson.plugins.timestamper.pipeline/PipelineTest/pct_timestamper_2_190_x___globalDecoratorAnnotator/), so I started looking into this failure. The 500 Server Error was caused by this stack trace:

```
Caused by: java.lang.NoClassDefFoundError: com/trilead/ssh2/crypto/Base64
	at org.jenkinsci.plugins.workflow.log.ConsoleAnnotators.setAnnotator(ConsoleAnnotators.java:101)
	at org.jenkinsci.plugins.workflow.log.FileLogStorage$1.writeHtmlTo(FileLogStorage.java:248)
	at hudson.model.Run.writeLogTo(Run.java:1488)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.commons.jexl.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:258)
	at org.apache.commons.jexl.parser.ASTMethod.execute(ASTMethod.java:104)
	at org.apache.commons.jexl.parser.ASTReference.execute(ASTReference.java:83)
	at org.apache.commons.jexl.parser.ASTReference.value(ASTReference.java:57)
	at org.apache.commons.jexl.parser.ASTReferenceExpression.value(ASTReferenceExpression.java:51)
	at org.apache.commons.jexl.ExpressionImpl.evaluate(ExpressionImpl.java:80)
	at hudson.ExpressionFactory2$JexlExpression.evaluate(ExpressionFactory2.java:74)
	at org.apache.commons.jelly.parser.EscapingExpression.evaluate(EscapingExpression.java:24)
	at org.apache.commons.jelly.impl.ExpressionScript.run(ExpressionScript.java:66)
	at org.apache.commons.jelly.TagSupport.invokeBody(TagSupport.java:161)
	at org.apache.commons.jelly.tags.core.WhitespaceTag.doTag(WhitespaceTag.java:48)
	at org.apache.commons.jelly.impl.TagScript.run(TagScript.java:269)
	... 115 more
```

Reading [the changelog](https://jenkins.io/changelog-stable/#v2.190.1), I see the following:

>Remove Trilead SSH library from Jenkins core and make it available in a new [detached plugin](https://plugins.jenkins.io/trilead-api). ([issue 43610](https://issues.jenkins-ci.org/browse/JENKINS-43610))

I'm not quite sure what to make of all of this yet, perhaps because I am unfamiliar with how split plugins work. I read [this documentation](https://wiki.jenkins.io/display/JENKINS/Split+plugin+from+core), which led me to [`split-plugins.txt`](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/jenkins/split-plugins.txt). Then I noticed that `prep.sh` is creating a `split-plugins.txt` whose contents are just the single line `# nothing`. Presumably this is to prevent `DetachedPluginsUtil.getImpliedDependencies` from returning any implied dependencies in order to simplify the implementation of the BOM somehow.

I set aside any latent questions about why `split-plugins.txt` is empty and decided to try adding `trilead-api` to the managed set explicitly, first at its latest version (1.0.5) and then with the version I'm currently using in production (1.0.3) with code like this:

```
diff --git a/bom-latest/pom.xml b/bom-latest/pom.xml
index a00b981..93a2ea6 100644
--- a/bom-latest/pom.xml
+++ b/bom-latest/pom.xml
@@ -253,6 +253,11 @@
                 <artifactId>timestamper</artifactId>
                 <version>1.10</version>
             </dependency>
+            <dependency>
+                <groupId>org.jenkins-ci.plugins</groupId>
+                <artifactId>trilead-api</artifactId>
+                <version>1.0.3</version>
+            </dependency>
             <dependency>
                 <groupId>org.jenkins-ci.plugins</groupId>
                 <artifactId>token-macro</artifactId>
```

This triggered the following error while building the sample project:

```
Managed dependencies of MavenProject: io.jenkins.tools.bom:sample:4-SNAPSHOT @ /home/basil/work/third-party/jenkinsci/bom/sample-plugin/target/sample-4-SNAPSHOT.pom: [com.github.spotbugs:spotbugs-annotations:jar, net.jcip:jcip-annotations:jar, org.jenkins-ci.main:jenkins-core:jar, org.jenkins-ci.main:jenkins-war:executable-war, org.jenkins-ci.main:jenkins-test-harness:jar, org.jenkins-ci:test-annotations:jar, junit:junit:jar, javax.servlet:javax.servlet-api:jar, org.codehaus.mojo:animal-sniffer-annotations:jar, org.codehaus.mojo.signature:java15:signature, org.codehaus.mojo.signature:java16:signature, org.codehaus.mojo.signature:java17:signature, org.codehaus.mojo.signature:java18:signature, commons-logging:commons-logging:jar, log4j:log4j:jar, org.mockito:mockito-core:jar, org.powermock:powermock-module-junit4:jar, org.powermock:powermock-api-mockito2:jar, org.objenesis:objenesis:jar, io.jenkins:configuration-as-code:jar, io.jenkins:configuration-as-code:jar:tests, org.jenkins-ci.plugins.workflow:workflow-api:jar, org.jenkins-ci.plugins.workflow:workflow-api:jar:tests, org.jenkins-ci.plugins.workflow:workflow-basic-steps:jar, org.jenkins-ci.plugins.workflow:workflow-cps:jar, org.jenkins-ci.plugins.workflow:workflow-cps:jar:tests, org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:jar, org.jenkins-ci.plugins.workflow:workflow-durable-task-step:jar, org.jenkins-ci.plugins.workflow:workflow-job:jar, org.jenkins-ci.plugins.workflow:workflow-scm-step:jar, org.jenkins-ci.plugins.workflow:workflow-step-api:jar, org.jenkins-ci.plugins.workflow:workflow-step-api:jar:tests, org.jenkins-ci.plugins.workflow:workflow-support:jar, org.jenkins-ci.plugins.workflow:workflow-support:jar:tests, org.jenkins-ci.plugins:ansicolor:jar, org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:jar, org.jenkins-ci.plugins:branch-api:jar, org.jenkins-ci.plugins:cloudbees-folder:jar, org.jenkins-ci.plugins:command-launcher:jar, org.jenkins-ci.plugins:credentials:jar, org.jenkins-ci.plugins:credentials-binding:jar, org.jenkins-ci.plugins:display-url-api:jar, org.jenkins-ci.plugins:durable-task:jar, org.jenkins-ci.plugins:git:jar, org.jenkins-ci.plugins:git:jar:tests, org.jenkins-ci.plugins:git-client:jar, org.jenkins-ci.plugins:git-server:jar, org.jenkins-ci.plugins:jdk-tool:jar, org.jenkins-ci.plugins:jquery:jar, org.jenkins-ci.plugins:jsch:jar, org.jenkins-ci.plugins:junit:jar, org.jenkins-ci.plugins:junit-attachments:jar, org.jenkins-ci.plugins:mailer:jar, org.jenkins-ci.plugins:matrix-project:jar, org.jenkins-ci.plugins:pipeline-build-step:jar, org.jenkins-ci.plugins:pipeline-input-step:jar, org.jenkins-ci.plugins:plain-credentials:jar, org.jenkins-ci.plugins:scm-api:jar, org.jenkins-ci.plugins:scm-api:jar:tests, org.jenkins-ci.plugins:script-security:jar, org.jenkins-ci.plugins:ssh-credentials:jar, org.jenkins-ci.plugins:ssh-slaves:jar, org.jenkins-ci.plugins:structs:jar, org.jenkins-ci.plugins:timestamper:jar, org.jenkins-ci.plugins:trilead-api:jar, org.jenkins-ci.plugins:token-macro:jar, org.jenkins-ci.plugins:variant:jar, org.jenkins-ci.ui:ace-editor:jar, org.jenkins-ci.ui:jquery-detached:jar, org.jenkins-ci:symbol-annotation:jar]
Do not see managed dependency org.codehaus.mojo.signature:java15
Do not see managed dependency org.codehaus.mojo.signature:java16
Do not see managed dependency org.codehaus.mojo.signature:java17
Do not see managed dependency org.codehaus.mojo.signature:java18
Do not see managed dependency org.mockito:mockito-core
Do not see managed dependency org.powermock:powermock-module-junit4
Do not see managed dependency org.powermock:powermock-api-mockito2
Do not see managed dependency org.objenesis:objenesis
```

Now here's where I am starting to get really confused. Running `mvn dependency:tree` in `trilead-api` shows that `trilead-api` doesn't consume any of the above libraries. The only reference I can find to them is in `plugin-pom`. In any case, they seem like test libraries and it doesn't seem right to add them to the managed set at first blush.

I decided to investigate whether my suspicion was correct, so I started reading `sample-plugin/check.groovy`. I see that we're failing in the first 18 lines of that file, but I wasn't able to quickly figure out the intention of the code in order to confirm whether it was doing what it was supposed to or not. In general, that file remains a bit opaque to me, since there aren't too many high-level comments and I haven't yet had the time to really dive into what it's doing.

Before I dive into `sample-plugin/check.groovy`, I was wondering if I'm on the right track here at all or if I should be going in some other direction. A quick check at this point would probably save me some time if I need to turn around.

**Comment by jglick** [comment]
I already started trying to deal with the Trilead split, for another plugin. Might not have pushed that yet. I have some ideas, but anyway CI is down again so this is on the back burner.

**Comment by jglick** [comment]
Also occurs to me that deleting all record of plugin splits as in https://github.com/jenkinsci/bom/blob/6d7fa6e5feb2b80ca16902b78f734f311705bd05/prep.sh#L31-L34 may be unsustainable. I _do_ want to have BOM-managed plugins override the ones bundled in `WEB-INF/detached-plugins/*.hpi`.

**Comment by basil** [comment]
> I _do_ want to have BOM-managed plugins override the ones bundled in `WEB-INF/detached-plugins/*.hpi`

That makes sense to me. I suppose that points to an implementation where we remove from `split-plugins.txt` only the entries that are not explicitly managed by the BOM (as opposed to all entries)?

**Comment by jglick** [comment]
> we remove from `split-plugins.txt` only the entries that are not explicitly managed by the BOM (as opposed to all entries)

Yes, plus updating the actual `*.hpi` for those that remain.

**Comment by timja** [comment]
This will fix the test failure in the last build:
https://github.com/jenkinsci/plain-credentials-plugin/pull/19

**Comment by timja** [comment]
Next failure is:
https://github.com/jenkinsci/jsch-plugin/pull/11

Turning off failFast so can see if there's any other errors

**Comment by timja** [comment]
Few other plugins to take a look at will hopefully take a look this morning

**Comment by timja** [comment]
Last build failed with lots of connection timed out to update center

**Comment by jglick** [comment]
> Last build failed with lots of connection timed out to update center

https://github.com/jenkinsci/bom/pull/189#issuecomment-585733325 ?

**Comment by timja** [comment]
```
[2020-02-24T08:06:33.956Z] Exception in thread "main" java.lang.RuntimeException: Invalid update center url : https://updates.jenkins-ci.org/update-center.json

[2020-02-24T08:06:33.956Z] 	at org.jenkins.tools.test.PluginCompatTester.extractUpdateCenterData(PluginCompatTester.java:586)

[2020-02-24T08:06:33.956Z] 	at org.jenkins.tools.test.PluginCompatTester.testPlugins(PluginCompatTester.java:184)

[2020-02-24T08:06:33.956Z] 	at org.jenkins.tools.test.PluginCompatTesterCli.main(PluginCompatTesterCli.java:163)

[2020-02-24T08:06:33.956Z] Caused by: java.net.ConnectException: Connection timed out (Connection timed out)

[2020-02-24T08:06:33.956Z] 	at java.net.PlainSocketImpl.socketConnect(Native Method)

[2020-02-24T08:06:33.956Z] 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)

[2020-02-24T08:06:33.956Z] 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)

[2020-02-24T08:06:33.956Z] 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)

[2020-02-24T08:06:33.956Z] 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)

[2020-02-24T08:06:33.956Z] 	at java.net.Socket.connect(Socket.java:607)

[2020-02-24T08:06:33.956Z] 	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:666)

[2020-02-24T08:06:33.956Z] 	at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)

[2020-02-24T08:06:33.956Z] 	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1162)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1056)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1570)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498)

[2020-02-24T08:06:33.956Z] 	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:268)

[2020-02-24T08:06:33.956Z] 	at java.net.URL.openStream(URL.java:1067)

[2020-02-24T08:06:33.956Z] 	at org.jenkins.tools.test.PluginCompatTester.extractUpdateCenterData(PluginCompatTester.java:584)

[2020-02-24T08:06:33.956Z] 	... 2 more
```

**Comment by jglick** [comment]
Yeah I am looking into it. The mode in which PCT is used in this project (`-war`) should never result in this code being run. Something is amiss.

**Comment by jglick** [comment]
https://github.com/jenkinsci/plugin-compat-tester/pull/169 seems to have been responsible. I am looking into whether this can be done better.

**Comment by jglick** [comment]
https://github.com/jenkinsci/plugin-compat-tester/pull/228

**Comment by timja** [comment]
Different pipeline tests are failing on every build,

mostly to do with assertLogContains failures:

```
Expected: a string containing "broken condition"
     but: was "Started
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] Start of Pipeline
[Pipeline] waitUntil
[Pipeline] {
[Pipeline] semaphore
[Pipeline] }
Will try again after 0.25 sec
[Pipeline] {
[Pipeline] semaphore
[Pipeline] }
[Pipeline] // waitUntil
[Pipeline] End of Pipeline
```

**Comment by jglick** [comment]
> mostly to do with assertLogContains failures

As in https://github.com/jenkinsci/workflow-job-plugin/pull/131#discussion_r292131627 these should be fixable by replacing `assertLogContains` with `waitForMessage`.

**Comment by timja** [comment]
@MarkEWaite 

This test failed here (first time I've seen it fail)
`testChooseGitRevisionToBuildByShaHash – hudson.plugins.git.util.DefaultBuildChooserTest`

https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-110/41/tests

<details>

<summary>Git test failed in last run </summary>

Stacktrace
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at hudson.plugins.git.util.DefaultBuildChooserTest.testChooseGitRevisionToBuildByShaHash(DefaultBuildChooserTest.java:31)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Standard Output
[junit2559266096640749226] $ git version
git version 2.20.1
[junit4262052903510475084] $ git init
Initialized empty Git repository in /home/jenkins/workspace/Tools_bom_PR-110/pct-work/git/target/tmp/junit2559266096640749226/junit4262052903510475084/.git/
[junit4262052903510475084] $ git add file
[junit4262052903510475084] $ git config user.name "Git SampleRepoRule"
[junit4262052903510475084] $ git config user.email gits@mplereporule
[junit4262052903510475084] $ git commit --message=init
[master (root-commit) 03e1564] init
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 file

</details>

**Comment by MarkEWaite** [comment]
@timja I don't know what caused that one time failure. I won't investigate it for now because I need to focus on the SCALE conference that I'm attending, reviewing other proposals for the git plugin, and fixing the git plugin documentation URL in the pom.

**Comment by Jetersen** [comment]
Uhh a green build :octocat:

**Comment by timja** [comment]
@jglick you able to take a look and check you're happy with the tests I ignored for this to work?

I'll adjust the PRs to the flakey components soon,

and I don't know how to fix the cloudbees-folder or trilead-api issues, but I moved them to ignored so we can hopefully move on with our lives

**Comment by jglick** [comment]
Huh, seems I cannot request myself as a reviewer on my own PR…

**Comment by jglick** [comment]
(I should have closed this and refiled as an origin branch PR as soon as I stopped active work on it, I guess.)

**Comment by timja** [comment]
maybe that will do ^ otherwise I can re-submit if you want

**Comment by jglick** [comment]
I will try to deal with this soon.

**Comment by timja** [comment]
> I will try to deal with this soon.

ping 😄 😉

**Comment by bitwiseman** [comment]
@jglick  Any thing to do here?

**Comment by jglick** [comment]
Yes, I need to review it and push out a release.

**Comment by jglick** [comment]
> Tests could be fixed later

No. Without green test runs we have no way of knowing what is or is not safe to merge here.

File: pct.sh [hunk 1]
```diff
@@ -37,23 +37,47 @@ then
...
+# https://github.com/jenkinsci/workflow-support-plugin/pull/105
+# https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/130
+# https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/110
+rm -fv pct-work/workflow-basic-steps/target/surefire-reports/TEST-org.jenkinsci.plugins.workflow.steps.CatchErrorStepTest.xml
+rm -fv pct-work/workflow-basic-steps/target/surefire-reports/TEST-org.jenkinsci.plugins.workflow.steps.WaitForConditionStepTest.xml
+rm -fv pct-work/workflow-basic-steps/target/surefire-reports/TEST-org.jenkinsci.plugins.workflow.support.steps.stash.StashTest.xml
+rm -fv pct-work/workflow-support/target/surefire-reports/TEST-org.jenkinsci.plugins.workflow.support.pickles.serialization.SerializationSecurityTest.xml
+rm -fv pct-work/workflow-durable-task-step/target/surefire-reports/TEST-org.jenkinsci.plugins.workflow.steps.durable_task.ShellStepTest.xml
+rm -fv pct-work/workflow-durable-task-step/target/surefire-reports/TEST-org.jenkinsci.plugins.workflow.support.steps.ExecutorStepTest.xml
+rm -fv pct-work/workflow-cps/target/surefire-reports/TEST-org.jenkinsci.plugins.workflow.cps.FlowDurabilityTest.xml
```
Comment (jglick) [inline pct.sh:72]: @timja I do see flakes here but did you try to track this down?

Comment (jglick) [reply pct.sh:72]:
```
FlowDurabilityTest.testCompleteAndLoadBuilds expected:<...ne] End of Pipeline
[]> but was:<...ne] End of Pipeline
[Finished: SUCCESS
]>
```

specifically

Comment (timja) [reply pct.sh:72]: Not sure it’s been a long time since I looked at these flakes

**Pull request merged**